doc/crypto/api/keys/management.rst

```diff
@@ -497,30 +496,38 @@ When creating a key, the attributes for the new key are specified in a `psa_key_
     That is, a key with a lifetime persistence level of `PSA_KEY_PERSISTENCE_VOLATILE`.
 
-    The key's policy and type are taken from ``attributes``.
-    The attributes can optionally specify a key size; in this case it must match the size determined from the key material.
-    A key size of ``0`` in ``attributes`` --- the default value --- indicates that the key size is determined by the key material.
+    Depending on the key being regsitered, the implementation can provide some or all of the key type, size, and policy.
+    For example:
 
-    The provided attributes must accurately describe the key.
-    It is recommended that the implementation verifies that the key material is compatible with the provided attributes; but it is the application's responsibility to ensure correctness.
+    *   Provisioned key material has a fixed size.
+        The implementation might permit the application to define the key type and policy, as long as these are compatible with the key material.
+    *   An implementation-specific derived key can require the application to provide a key type and size, using these in the derivation process.
+    *   An implementation-provided key can be fully defined by the implementation, with a fixed type, size, and policy.
+        The call to `psa_attach_key()` needs to specify the location and label of the key, and a macthing policy, in order to obtain a key id.
 
-    Calling `psa_destroy_key()` with a key identifier returned by `psa_attach_key()` will remove the key identifer and policy from the key store, but the original key material remains within the implementation.
+    Calling `psa_destroy_key()` with a key identifier returned by `psa_attach_key()` will remove the key identifer and policy from the key store, but any implementation-provided key material remains within the implementation.
+    A subsequent call to `psa_attach_key()` with the same parameters will return a new key identifier for the same key.
 
-    It is :scterm:`implementation defined` whether the same key material can be attached to multiple key identifiers concurrently.
+    It is :scterm:`implementation defined` whether the same implementation-provided key can be attached to multiple key identifiers concurrently.
 
     .. note::
 
-        This function is intended for scenarios where key material is provisioned outside the |API|, and the application needs to use such keys within the API framework.
+        This function is intended for scenarios where key material is provided outside the |API|, and the application needs to use such keys within the |API| framework.
 
-        Implementations may impose restrictions on which keys can be registered, depending on their storage architecture and security policies.
-
-        The function does not allow registering keys that are not already present in the implementation's storage.
+        The function does not allow registering keys that are not already provided by the implementation.
```

Comment:
I find this sentence hard to understand. The _key material_ is provided by the implementation but the _key object_ isn't. Also, there's a double negation, which is rarely easy to read. Suggestion:

> The function only allows registering key material that is provided by the implementation. To import new key material, use `psa_import_key()`.

Reply:
Yes, removing the double negative is much clearer.